lib/redis/client.rb

```diff
@@ -595,11 +596,6 @@ def sentinel_detect
           end
 
           raise CannotConnectError, "No sentinels available."
-        rescue Redis::CommandError => err
```

Comment:
We can remove this error handling as workaround because the `password` option is separated.